src/components/InputTypeCountry.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import StyledSelect from './StyledSelect';
+
+class InputTypeCountry extends Component {
+  static propTypes = {
+    options: PropTypes.arrayOf(
+      PropTypes.shape({
+        country: PropTypes.string.isRequired,
+        abbreviation: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+    name: PropTypes.string,
+    onChange: PropTypes.func,
+    defaultValue: PropTypes.string,
+  };
+
+  state = {
+    error: false,
+  };
+
+  handleChange = ({ value }) => {
+    if (value === null) {
+      this.setState({
+        error: true,
+      });
+    } else {
+      this.setState({
+        error: false,
+      });
+      this.props.onChange(value);
+    }
+  };
+
+  getCountryName(countryISO) {
+    const { options } = this.props;
+    const { country, abbreviation } = options.find(({ abbreviation }) => abbreviation === countryISO);
```

Comment:
The name `abbreviation` is surprising, it's usually named `code` or similar.

Reply:
:slightly_smiling_face: Could have used the IDE  to replace `abbreviation` as `code` but thought about keeping it original as the file I got it from.